src/solver.f90

```diff
@@ -397,56 +398,83 @@ subroutine curl(self, o_x, o_y, o_z, u, v, w)
       class(field_t), intent(inout) :: o_x, o_y, o_z !! omega_x/_y/_z
```

Comment:
Confusing thing here is we normaly use `_x/y/z` to indicate the orientation of the array, but curl gives us back a field with three component by definition, and `o_x/y/z` here indicate three components of this vector. If it sounds better to you I can rename as `_i/j/k` so that it indicates the individual components of a vector field.

Reply:
_Xcomp / _Ycomp / _Zcomp?

Reply:
I think go for the `_i/j/k` or even a combination of both to indicate the index and the orientation, e.g.
```
oi_x
oj_x
ok_x
```
or whichever orientation is appropriate

Reply:
How about `_xhat, _yhat, _zhat`? `comp` sounds like the shorthand for compact

Reply:
Xpart / Ypart / Zpart?

Reply:
I tend to omit `_x` for x oriented arrays in most places because its the default. Its like we omit `_v2v` if the operator works on velocity grid.